server/src/main/java/org/elasticsearch/action/support/broadcast/node/TransportBroadcastByNodeAction.java

```diff
@@ -308,13 +309,17 @@ protected void sendItemRequest(Map.Entry<String, List<ShardRouting>> entry, Acti
                     nodeRequest.setParentTask(clusterService.localNode().getId(), task.getId());
                 }
 
-                transportService.sendRequest(
-                    node,
-                    transportNodeBroadcastAction,
-                    nodeRequest,
-                    transportRequestOptions,
-                    new ActionListenerResponseHandler<>(listener, nodeResponseReader, executor)
-                );
+                try {
```

Comment:
Just a thought that popped into my head: Having these try-finally is quite annoying but hard to change right now. It seems the Netty approach of counting down in whatever method you call to send something would be a nice refactoring here to save lots of cod :)

Reply:
I _vastly_ prefer having the incref/decref pairs right next to each other so you can see there's no leaks when reviewing changes. Otherwise you have to keep track of which methods do/don't implicitly decref their args, and make sure to do the decref on all possible return paths (which change over time). Netty's approach is cute, but in practice it's a great way to breed bugs IMO.